File: zig/armstrong-numbers/armstrong_numbers.zig
const std = @import("std");
const math = std.math;

pub fn isArmstrongNumber(num: u128) bool {
    var tmpNum: u128 = num;
    var count: u128 = 0;
    while (tmpNum > 0) {
        tmpNum /= 10;
        count += 1;
    }
    tmpNum = num;
    var sum: u129 = 0;
    while (tmpNum > 0): (tmpNum /= 10) {
        var digit = tmpNum % 10;
        sum += math.pow(u128, digit, count);
    }
    return num == sum;
}

File: cpp/armstrong-numbers/armstrong_numbers.cpp
#include "armstrong_numbers.h"

namespace armstrong_numbers {
  bool is_armstrong_number(int num) {

    int numDigits = armstrong_numbers::numberDigits(num);
    int result = 0;
    int tmpNum = num;
    while (tmpNum) {
        int curNum = tmpNum % 10;   
        result += pow(curNum, numDigits);
        tmpNum /= 10;
    }

    return result == num;
  }

  int numberDigits(int number) {
    int digits = 0;
    if (number < 0) digits = 1; // remove this line if '-' counts as a digit
    while (number) {
        number /= 10;
        digits++;
    }
    return digits;
  }
}
